python/mxnet/model.py

```diff
@@ -113,25 +127,36 @@ def _update_params_on_kvstore(param_arrays, grad_arrays, kvstore, param_names):
         # push gradient, priority is negative index
         kvstore.push(name, grad_list, priority=-index)
         # pull back the weights
-        kvstore.pull(name, arg_list, priority=-index)
+        if _contains_non_default_storage(arg_list):
```

Comment:
Right. For distributed training we want to reduce the network traffic and use `row_sparse_pull` instead. We may change it if it's not user-friendly. One distributed trainig example with sparse weight is here: https://github.com/eric-haibin-lin/mxnet/blob/sparse/example/sparse/linear_classification.py#L171